tests/loggers/test_neptune.py

```diff
@@ -29,6 +29,8 @@ def test_neptune_online(neptune):
     # It's important to check if the internal variable _experiment was initialized in __init__.
     # Calling logger.experiment would cause a side-effect of initializing _experiment,
     # if it wasn't already initialized.
+    assert logger._experiment is None
+    _ = logger.experiment
     assert logger._experiment == created_experiment
     assert logger.name == created_experiment.name
     assert logger.version == created_experiment.id
```

Comment:
@psinger I rebased the branch and updated the tests so that they pass with the change you made in neptune. 
When doing that, I saw this comment in the test. I'm not sure what this is about. I see no evidence that we are forced to initialize the neptune experiment at init. How do you see it?